cypress/plugins/post_message_as.js

```diff
@@ -1,35 +1,40 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-require('@babel/polyfill');
-require('isomorphic-fetch');
-
-const {Client4} = require('mattermost-redux/client');
 const axios = require('axios');
 
 const cypressConfig = require('../../cypress.json');
 
 module.exports = async ({sender, message, channelId, createAt = 0}) => {
-    const url = `${cypressConfig.baseUrl}/api/v4/users/login`;
-
-    const response = await axios({
-        url,
+    const loginResponse = await axios({
+        url: `${cypressConfig.baseUrl}/api/v4/users/login`,
         headers: {'X-Requested-With': 'XMLHttpRequest'},
         method: 'post',
         data: {login_id: sender.username, password: sender.password},
     });
 
-    const token = response.headers.token;
-
-    Client4.setUrl(cypressConfig.baseUrl);
-    Client4.setToken(token);
+    const setCookie = loginResponse.headers['set-cookie'];
+    let cookieString = '';
+    setCookie.forEach((cookie) => {
+        const nameAndValue = cookie.split(';')[0];
+        cookieString += nameAndValue + ';';
+    });
 
-    const options = {
-        channel_id: channelId,
-        message,
-        type: '',
-        create_at: createAt,
-    };
+    const response = await axios({
+        url: `${cypressConfig.baseUrl}/api/v4/posts`,
+        headers: {
+            'Content-Type': 'application/json',
+            'X-Requested-With': 'XMLHttpRequest',
+            Cookie: cookieString,
+        },
+        method: 'post',
+        data: {
+            channel_id: channelId,
+            message,
+            type: '',
+            create_at: createAt,
+        },
+    });
 
-    return Client4.createPost(options);
+    return {status: response.status, data: response.data, error: response.error};
```

Comment:
Not totally sure why [postMessageAs](https://build-push.internal.mattermost.com/job/mattermost-ui-testing/job/mattermost-cypress/74/artifact/mattermost-webapp/cypress/screenshots/at_mentions/at_mentions_spec.js/at-mention%20--%20N14572%20does%20not%20trigger%20notifications%20with%20channel-wide%20mentions%20unchecked%20(failed).png) is not working as expected.  I just removed the use of `Client4`; let's observe if it will pass on CI.

Reply:
mattermost-redux changes perhaps? I haven't looked at recent commits, but I think it's totally fine for us to use the API directly, ultimately that's what the client is doing. This way we have more control over the requests as well. 👍

Reply:
I don't think so as the current mattermost-redux is working locally but let's eliminate potential issue by going thru API call directly.